### PR TITLE
Added the compression-dictionary draft

### PR DESCRIPTION
**Pull request by pmeenan**
This is the -00 draft of the compression-dictionary ID.  I used the resumable-upload.md draft as a template for the metadata fields at the top of the document so hopefully they are correct for an IETF-standards track draft.

### CODE REVIEW

File: draft-ietf-httpbis-compression-dictionary.md [hunk 1]
```diff
@@ -0,0 +1,452 @@
+---
+title: Compression Dictionary Transport
+abbrev: compression-dictionary
+docname: draft-ietf-httpbis-compression-dictionary-latest
+category: std
+
+ipr: trust200902
```
Comment (martinthomson) [inline draft-ietf-httpbis-compression-dictionary.md:7]: You can replace this with:

```suggestion
consensus: true
v: 3
```

File: draft-ietf-httpbis-compression-dictionary.md [hunk 2]
```diff
@@ -0,0 +1,452 @@
...
+area: ART
+workgroup: HTTP
+keyword: Internet-Draft
+
+stand_alone: yes
+smart_quotes: no
+pi: [toc, sortrefs, symrefs]
+
```
Comment (martinthomson) [inline draft-ietf-httpbis-compression-dictionary.md:15]: And drop this part.

```suggestion
```

File: draft-ietf-httpbis-compression-dictionary.md [hunk 3]
```diff
@@ -0,0 +1,452 @@
+---
+title: Compression Dictionary Transport
+abbrev: compression-dictionary
+docname: draft-ietf-httpbis-compression-dictionary-latest
+category: std
+
+ipr: trust200902
+area: ART
+workgroup: HTTP
+keyword: Internet-Draft
```
Comment (martinthomson) [inline draft-ietf-httpbis-compression-dictionary.md:10]: Consider some keywords here:

```suggestion
keyword:
  - blockchain
  - climate change
  - flat earth
```

File: draft-ietf-httpbis-compression-dictionary.md [hunk 4]
```diff
@@ -0,0 +1,452 @@
...
+  -
+    ins: Y. Weiss
+    name: Yoav Weiss
+    role: editor
+    organization: Google LLC
+    email: yoavweiss@google.com
+
+normative:
+  FOLDING: RFC8792
+  RFC9110:  # HTTP semantics
```
Comment (martinthomson) [inline draft-ietf-httpbis-compression-dictionary.md:41]: This should be called HTTP:

```suggestion
  HTTP: RFC9110
```

File: draft-ietf-httpbis-compression-dictionary.md [hunk 5]
```diff
@@ -0,0 +1,452 @@
...
+
+informative:
+  HTTP: RFC7230
```
Comment (martinthomson) [inline draft-ietf-httpbis-compression-dictionary.md:44]: And this shouldn't be used.

```suggestion
```

File: draft-ietf-httpbis-compression-dictionary.md [hunk 6]
```diff
@@ -0,0 +1,452 @@
...
+  Origin: RFC6454
+  ABNF: RFC5234
+  RFC3986:  # URL
+  RFC8941:  # Structured fields
```
Comment (martinthomson) [inline draft-ietf-httpbis-compression-dictionary.md:48]: See https://httpwg.org/admin/editors/style-guide#reference-style

```suggestion
  STRUCTURED-FIELDS: RFC8941
```

File: draft-ietf-httpbis-compression-dictionary.md [hunk 7]
```diff
@@ -0,0 +1,452 @@
...
+
+normative:
+  FOLDING: RFC8792
+  RFC9110:  # HTTP semantics
+
+informative:
+  HTTP: RFC7230
+  Origin: RFC6454
+  ABNF: RFC5234
+  RFC3986:  # URL
```
Comment (martinthomson) [inline draft-ietf-httpbis-compression-dictionary.md:47]:
```suggestion
  URL: RFC3986
```

File: draft-ietf-httpbis-compression-dictionary.md [hunk 8]
```diff
@@ -0,0 +1,452 @@
...
+that support using external dictionaries (e.g. Brotli {{RFC7932}} and Zstandard
+{{RFC8878}}).
+
+--- middle
+
+# Introduction
+
+This specification defines a mechanism for using designated {{HTTP}} responses
+as an external dictionary for future HTTP responses for compression schemes
+that support using external dictionaries (e.g. Brotli {{RFC7932}} and Zstandard
```
Comment (martinthomson) [inline draft-ietf-httpbis-compression-dictionary.md:66]:
```suggestion
that support using external dictionaries (e.g., Brotli {{RFC7932}} and Zstandard
```

File: draft-ietf-httpbis-compression-dictionary.md [hunk 9]
```diff
@@ -0,0 +1,452 @@
...
+  RFC8941:  # Structured fields
+  RFC4648:  # Base16 encoding
+  RFC7932:  # Brotli
+  RFC8878:  # Zstandard
+
+--- abstract
+
+This specification defines a mechanism for using designated {{HTTP}} responses
+as an external dictionary for future HTTP responses for compression schemes
+that support using external dictionaries (e.g. Brotli {{RFC7932}} and Zstandard
```
Comment (martinthomson) [inline draft-ietf-httpbis-compression-dictionary.md:57]:
```suggestion
that support using external dictionaries (e.g., Brotli {{RFC7932}} and Zstandard
```

File: draft-ietf-httpbis-compression-dictionary.md (hunk 1 above)
Comment (pmeenan) [reply draft-ietf-httpbis-compression-dictionary.md]: Done, thx

File: draft-ietf-httpbis-compression-dictionary.md (hunk 2 above)
Comment (pmeenan) [reply draft-ietf-httpbis-compression-dictionary.md]: Done, thx

File: draft-ietf-httpbis-compression-dictionary.md (hunk 3 above)
Comment (pmeenan) [reply draft-ietf-httpbis-compression-dictionary.md]: Done, thx

File: draft-ietf-httpbis-compression-dictionary.md (hunk 4 above)
Comment (pmeenan) [reply draft-ietf-httpbis-compression-dictionary.md]: Done, thx

File: draft-ietf-httpbis-compression-dictionary.md (hunk 5 above)
Comment (pmeenan) [reply draft-ietf-httpbis-compression-dictionary.md]: Done, thx

File: draft-ietf-httpbis-compression-dictionary.md (hunk 6 above)
Comment (pmeenan) [reply draft-ietf-httpbis-compression-dictionary.md]: Done, thx

File: draft-ietf-httpbis-compression-dictionary.md (hunk 7 above)
Comment (pmeenan) [reply draft-ietf-httpbis-compression-dictionary.md]: Done, thx

File: draft-ietf-httpbis-compression-dictionary.md (hunk 8 above)
Comment (pmeenan) [reply draft-ietf-httpbis-compression-dictionary.md]: Done, thx

File: draft-ietf-httpbis-compression-dictionary.md (hunk 9 above)
Comment (pmeenan) [reply draft-ietf-httpbis-compression-dictionary.md]: Done, thx

**Comment by pmeenan** [comment]
Thanks for the edits (hope it didn't generate too many emails as I merged the changes).

**Comment by martinthomson** [comment]
> hope it didn't generate too many emails as I merged the changes

There were a few.  From the "Files" panel, you can batch the suggestions and apply them as a single commit.

**Comment by martinthomson** [comment]
Another point: @mnot or @tfpauly probably need to give @pmeenan (and @yoavweiss) access to this repo.

**Comment by tfpauly** [comment]
Invited @pmeenan . @yoavweiss already was on the list

File: draft-ietf-httpbis-compression-dictionary.md [hunk 10]
```diff
@@ -0,0 +1,452 @@
...
+~~~ http-message
+Vary: accept-encoding, available-dictionary
+~~~
+
+# IANA Considerations
+
+## Content Encoding
+
+IANA is asked to update the "HTTP Content Coding Registry" registry
+({{RFC9110}}) according to the table below:
```
Comment (mnot) [inline draft-ietf-httpbis-compression-dictionary.md:311]:
```suggestion
({{HTTP}}) according to the table below:
```

File: draft-ietf-httpbis-compression-dictionary.md [hunk 11]
```diff
@@ -0,0 +1,452 @@
...
+|--------|---------------------------------------------------------------------------------------|-------------|
+| br-d   | A stream of bytes compressed using the Brotli protocol with an external dictionary    | {{RFC7932}} |
+| zstd-d | A stream of bytes compressed using the Zstandard protocol with an external dictionary | {{RFC8878}} |
+|--------|---------------------------------------------------------------------------------------|-------------|
+
+## Header Field Registration
+
+IANA is asked to update the
+"Hypertext Transfer Protocol (HTTP) Field Name Registry" registry
+({{RFC9110}}) according to the table below:
```
Comment (mnot) [inline draft-ietf-httpbis-compression-dictionary.md:324]:
```suggestion
({{HTTP}}) according to the table below:
```

**Comment by mnot** [comment]
@pmeenan looks good to me FWIW - merge and release when you see fit. 

Note that you can release from github -- see:
  https://github.com/martinthomson/i-d-template/blob/main/doc/SUBMITTING.md#github-release

**Pull request merged**